src/main/haskell/kore/test/Test/Kore/SMT/SMT.hs

```diff
@@ -0,0 +1,173 @@
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+module Test.Kore.SMT.SMT where
+
+import Test.QuickCheck
+       ( Property, (===) )
+
+import           Data.Map
+import           Data.Reflection
+
+
+import           Kore.AST.PureML
+import           Kore.AST.Sentence
+import           Kore.AST.MetaOrObject
+import           Kore.ASTUtils.SmartConstructors
+import           Kore.ASTUtils.SmartPatterns
+import           Kore.Proof.Dummy
+
+
+import           Kore.IndexedModule.IndexedModule
+import           Kore.IndexedModule.MetadataTools
+import           Kore.SMT.SMT
+
+import           Test.Kore.Builtin.Int hiding 
+                 ( intModule, indexedModules )
+import           Test.Kore.Builtin.Bool 
+                 ( boolSort ) 
+
+indexedModules :: Map ModuleName (KoreIndexedModule SMTAttributes)
+indexedModules = verify intDefinition
+
+intModule :: KoreIndexedModule SMTAttributes
+Just intModule = Data.Map.lookup intModuleName indexedModules
+
+tools :: MetadataTools Object SMTAttributes
+tools = extractMetadataTools intModule
+
+vInt :: String -> CommonPurePattern Object
+vInt s = V $ s `varS` intSort
+
+a, b, c :: CommonPurePattern Object
+a = vInt "a"
+b = vInt "b"
+c = vInt "c"
+
+vBool :: String -> CommonPurePattern Object
+vBool s = V $ s `varS` boolSort
+
+p, q :: CommonPurePattern Object
+p = vBool "p"
+q = vBool "q"
+
+type Op = CommonPurePattern Object -> CommonPurePattern Object -> CommonPurePattern Object
+plus, sub, mul, div :: Op
+plus  a b = App_ addSymbol  [a, b]
+sub a b = App_ subSymbol  [a, b]
+mul a b = App_ mulSymbol  [a, b]
+div   a b = App_ tdivSymbol [a, b]
+
+prop_1 :: Property
+prop_1 = run prop1
+
+prop_2 :: Property
+prop_2 = run prop2
+
+prop_3 :: Property
+prop_3 = run prop3
+
+prop_4 :: Property
+prop_4 = run prop4
+
+prop_pierce :: Property
+prop_pierce = run propPierce
+
+prop_tdiv :: Property
+prop_tdiv = run propDiv
+
+prop_tmod :: Property
+prop_tmod = run propMod
+
+prop_demorgan :: Property
+prop_demorgan = run propDeMorgan
+
+prop_true :: Property
+prop_true = run propTrue
+
+prop_false :: Property
+prop_false = run propFalse
+
+run :: CommonPurePattern Object -> Property
+run prop = (give tools $ provePattern prop) === Just True
+
+prop1 :: CommonPurePattern Object
+prop1 = dummyEnvironment $ mkNot $ 
```

Comment:
Note that the prop_1 test is somewhat less readable than when inlining the definition for this.